File: Python/scraping/scrape_products_amazon.py
import json
from playwright.sync_api import sync_playwright

def main():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        page.goto('https://www.amazon.com/b?node=283155')
        page.wait_for_selector('.a-column')

        product_data = []
        products = page.query_selector_all('.a-column')

        for product in products:
            title = product.query_selector('.p13n-sc-truncate-desktop-type2')
            if title:
                rating = product.query_selector('.a-icon-alt')
                reviews_count = product.query_selector('.a-size-small')
                image = product.query_selector('img')

                product_data.append({
                    'title': title.inner_text() if title else None,
                    'rating': rating.inner_text() if rating else None,
                    'reviews_count': reviews_count.inner_text() if reviews_count else None,
                    'image_url': image.get_attribute('src') if image else None
                })

        with open('product_data.json', 'w', encoding='utf-8') as f:
            json.dump(product_data, f, ensure_ascii=False, indent=2)

        browser.close()

if __name__ == "__main__":
    main()

File: NodeJS/scraping/scrape_products_amazon.js
const { chromium } = require('playwright'); 
const fs = require('fs');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();
  await page.goto('https://www.amazon.com/b?node=283155');
  await page.waitForSelector('.a-column');

  const productData = await page.$$eval('.a-column', (products) =>
    products.map((product) => {
      const getText = (selector) => {
        const el = product.querySelector(selector);
        return el ? el.textContent.trim() : null;
      };

      const getSrc = (selector) => {
        const el = product.querySelector(selector);
        return el ? el.src : null;
      };

      return {
        title: getText('.p13n-sc-truncate-desktop-type2'),
        rating: getText('.a-icon-alt'),
        reviews_count: getText('.a-size-small'),
        image_url: getSrc('img'),
      };
    })
  );

  fs.writeFileSync('product_data.json', JSON.stringify(productData, null, 2), 'utf-8');

  await browser.close();
})();
